contracts/Removal.sol

```diff
@@ -80,37 +110,87 @@ contract Removal is
     return removalId.unpackRemovalIdV0();
   }
 
+  // TODO do we want a batch version of this?
+  /**
+   * @notice Get the restriction schedule id (which is the removal's project id) for a given removal id.
+   */
+  function getProjectIdForRemoval(uint256 removalId)
+    public
+    view
+    returns (uint256)
+  {
+    return _removalIdToProjectId[removalId];
+  }
+
+  /**
+   * @notice Get the restriction schedule data for a given project id.
+   */
+  function getScheduleData(uint256 projectId)
+    public
+    view
+    returns (ScheduleData memory)
+  {
+    return _projectIdToScheduleData[projectId];
+  }
+
   /**
    * @dev mints multiple removals at once (for a single supplier).
    * If `list` is true in the decoded BatchMintRemovalsData, also lists those removals for sale in the market.
    * amounts: [100 * (10 ** 18), 10 * (10 ** 18), 50 * (10 ** 18)] <- 100 tonnes, 10 tonnes, 50 tonnes in standard erc20 units (wei)
    * token id 0 URI points to vintage information (e.g., 2018) nori.com/api/removal/0 -> { amount: 100, supplier: 1, vintage: 2018, ... }
    * token id 1 URI points to vintage information (e.g., 2019) nori.com/api/removal/1 -> { amount: 10, supplier: 1, vintage: 2019, ... }
    * token id 2 URI points to vintage information (e.g., 2020) nori.com/api/removal/2 -> { amount: 50, supplier: 1, vintage: 2020, ... }
+   *
    * @param to The supplier address
    * @param amounts Each removal's tonnes of CO2 formatted as wei
    * @param ids The token ids to use for this batch of removals. The id itself encodes the supplier's ethereum address, a parcel identifier,
-   * the vintage, country code, state code, methodology identifer, and methodology version.
-   * @param data Encodes the market contract address and a unique identifier for the parcel from whence these removals came.
+   * the vintage, country code, state code, methodology identifer, methodology version, and id format.
+   * @param data Encodes the project id and schedule start time for this batch of removals, the market contract
+   * address and a boolean that indicates whether to list these removals for sale now.
    */
   function mintBatch(
     address to,
     uint256[] memory amounts,
     uint256[] memory ids,
     bytes memory data
   ) public override {
+    if (!(amounts.length == ids.length)) {
+      revert ArrayLengthMismatch({array1Name: "amounts", array2Name: "ids"});
```

Comment:
is the intent with this custom error to use it in more than one place? Seems like this could just as easily just be `revert AmountAndIdArrayLengthMismatch` (name could be improved) which would conserve some gas

Reply:
that's an idea... I was actually wondering about custom errors and wondering if there's a way to create a sort of library of them that are shared across the contracts, because I feel like I was re-writing similar ones in a few places.  not sure if that's worth it or if it's best to be as specific as possible and, as you mention here, avoid passing args, which I'm guessing is where the gas savings occur.

maybe we can consider this wholistically in the custom errors ticket where we just overhaul everything at once and make decisions about how much re-use to achieve?

Reply:
cool i'm going to add a note to that ticket to make sure we consider this